ert/data/_record.py

```diff
@@ -131,34 +137,71 @@ class BlobRecord(Record):
     data: blob_record_data
 
 
-class RecordCollection(_DataElement):
-    records: Union[Tuple[NumericalRecord, ...], Tuple[BlobRecord, ...]]
-    ensemble_size: Optional[int] = None
+_RecordTupleType = Union[Tuple[NumericalRecord, ...], Tuple[BlobRecord, ...]]
+
+
+class RecordCollection:
+    def __init__(self, records: Any, ensemble_size: Optional[PositiveInt] = None):
+        self._records: _RecordTupleType
+        try:
+            # Passing a single records indicates that a uniform record is created.
+            record = self._make_record(records)
+            if ensemble_size is None:
+                raise ValueError("Ensemble size missing for uniform record")
+            self._records = cast(_RecordTupleType, (record,) * ensemble_size)
+            self._ensemble_size = ensemble_size
+            self._is_uniform = True
+        except ValueError:
+            # If not a uniform record, it must be an iteratable of records.
+            self._records = cast(
+                _RecordTupleType, tuple(self._make_record(record) for record in records)
+            )
+            if len(self._records) < 1:
+                raise ValueError("At least one record must be provided")
+            for record in self._records[1:]:
+                if record.record_type != self._records[0].record_type:
+                    raise ValueError("Ensemble records must have a uniform record type")
+            self._ensemble_size = len(self._records)
+            if ensemble_size is not None and ensemble_size is not self._ensemble_size:
+                raise ValueError("Ensemble size does not match the record count")
+            self._is_uniform = False
+
+    def __eq__(self, other: object) -> bool:
+        if isinstance(other, RecordCollection):
+            return (
```

Comment:
Isn't this equivalent to `self.__dict__ == other.__dict__`?

Reply:
Probably. Is it preferable?

Reply:
Yes

Reply:
ah, yes, of course, guess I was sleeping when I wrote that code...